File: lib/dingen_handler_world.ex
defmodule Dingen.Handler.World do
  def init(_type, req, _opts) do
    {:ok, req, :no_state}
  end

  def handle(req, state) do
    headers = [{"content-type", "text/html"}]
    body = "
      <title>Hello world!</title>
      <p>Hello world!</p>
    "
    {:ok, req} = :cowboy_req.reply(200, headers, body, req)
    {:ok, req, state}
  end

  def terminate(_reason, _req, _state) do
    :ok
  end
end

File: lib/dingen_handler_world.erl
-module(dingen_handler_world).

-export([init/3, handle/2, terminate/3]).



init(_Type, Req, _Opts) ->
  {ok, Req, no_state}.


handle(Req, State) ->
  Headers = [{"content-type", "text/html"}],
  Body = "
    <title>Hello world!</title>
    <p>Hello world!</p>
  ",
  {ok, Req2} = cowboy_req:reply(200, Headers, Body, Req),
  {ok, Req2, State}.


terminate(_Reason, _Req, _State) ->
  ok.
